Add scene API for newton Kamino solver

vastsoun opened the pull request:
## Description
This PR addresses #15, by adding an initial prototype for Kamino-specific solver parameters in the form of a specialized applied scene schema  `NewtonKaminoSceneAPI` which extends the base `NewtonSceneAPI`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/newton-physics/newton-usd-schemas/blob/HEAD/CONTRIBUTING.md).

vastsoun commented:
Credits to @Milad-Rakhsha-NV, I basically based everything off of his PR for XPBD.

andrewkaufman commented on newton_usd_schemas/plugInfo.json:
```diff
@@ -58,6 +58,20 @@
                 "Material"
               ]
+            },
+            "NewtonPhysicsKaminoSceneAPI": {
```
Lets shuffle this up above CollisionAPI & just below XpbdAPI (after #18 is merged)

andrewkaufman commented on tests/test_solver_kamino.py:
```diff
@@ -0,0 +1,142 @@
+
+    def test_api_application(self):
+        self.scene.ApplyAPI("NewtonKaminoSceneAPI")
+        self.assertTrue(self.scene.HasAPI("NewtonKaminoSceneAPI"))
```
As in #18 lets add the extra check


```suggestion
        self.assertTrue(self.scene.HasAPI("NewtonSceneAPI"))
        self.assertTrue(self.scene.HasAPI("NewtonKaminoSceneAPI"))
```

andrewkaufman commented on tests/test_solver_kamino.py:
```diff
@@ -0,0 +1,142 @@
+        self.assertTrue(success)
+        self.assertAlmostEqual(attr.Get(), 1e-4, places=7)
+
+    def test_padmm_warmstart(self):
```
For the token attributes I'd suggest testing all allowed values & testing that a made up token fails to set

vastsoun replied:
Good point, will add them.

andrewkaufman replied:
ugg, this is an AI response, but its trained on USD so I believe it (also seems to match what we're finding here)

<img width="713" height="1128" alt="image" src="https://github.com/user-attachments/assets/b9125cc3-cecc-4c11-8633-9216b1ba9f66" />

andrewkaufman commented on newton_usd_schemas/generatedSchema.usda:
```diff
@@ -70,3 +70,125 @@ class NewtonMaterialAPI "NewtonMaterialAPI" (
+        """
+    )
+
+    uniform bool newton:kamino:padmm:acceleration = true (
```
You call this `use_acceleration` in the code, any reason to not match that here? e.g.

- `useAcceleration` camelCase version of the solver arg
- `accelerationEnabled` usd-esque bool style

I don't mind, just wondering

vastsoun replied:
I've still not yet settled on naming all the settings in the code itself, so I can remove this for now and add it back later.

andrewkaufman commented on newton_usd_schemas/generatedSchema.usda:
```diff
@@ -70,3 +70,125 @@ class NewtonMaterialAPI "NewtonMaterialAPI" (
+        """
+    )
+
+    uniform float newton:kamino:constraints:alpha = 0.01 (
```
I feel like its a little overkill at this stage. Just one KaminoAPI for the time being. Unless the solver landing will actually be called PADMM, then maybe `NewtonPADMMSceneAPI` instead?

vastsoun replied:
@Milad-Rakhsha-NV, these alpha,beta etc parameters are solver-specific and applied globally to all relevant constraints within the solvers' computations so I could rename them to `:padmm:` but it then wouldnt exactly reflect how they are in the code. Dunno, I'm also kinda split on this, but since we're not rushing to merge this we can leave it as is for now and decide later.

vastsoun commented on newton_usd_schemas/generatedSchema.usda:
```diff
@@ -70,3 +70,125 @@ class NewtonMaterialAPI "NewtonMaterialAPI" (
+
+        Range: (1e-8, 1e-2)"""
+    )
+
```
Yeah, those are TBD and still experimental. I've only added those that I'm confident will persist.

vastsoun commented on newton_usd_schemas/generatedSchema.usda:
```diff
@@ -70,3 +70,125 @@ class NewtonMaterialAPI "NewtonMaterialAPI" (
+        Range: [0.0, 1.0]"""
+    )
+
+    uniform float newton:kamino:constraints:delta = 1e-6 (
```
Sort of, but it's a solver-level tolerance on when to begin applying forces, while the `NewtonCollisionAPI.newton:contactMargin` effects collision detection itself. So I've found in practice both bring different benefits to making the simulation more robust and stable in terms of contact flickering.

vastsoun replied:
I removed to avoid the parameter concflict.

vastsoun commented on newton_usd_schemas/plugInfo.json:
```diff
@@ -58,6 +58,20 @@
                 "Material"
               ]
+            },
+            "NewtonPhysicsKaminoSceneAPI": {
```
Sure, no problem, I will revert this PR to draft for the time being, and just try and keep it alive in preparation for the eventual merge.

vastsoun commented:
@andrewkaufman  @Milad-Rakhsha-NV the PR is ready to review and merge. 

What's done:
- Rebased to main
- Revised Kamino scene attributes to remove conflicting and reduce further to those that won't change
- Added attribute limits 
- Incorporated majority of comments in latest commits, some we can revise further as part of the PR
- Fixed and updated unit tests

Remaining topics:
- If the namescopes are too deep, then I can just take the second last as prefix to attribute names, such as: `newton:kamino:padmmDualTolerance` instead of `newton:kamino:padmm:dualTolerance`.  I still prefer the existing ones though because I think they are cleaner and easier to read. If there is no impact on implementation, can we just leave them as is for now? The extra namescope could open the door (in the future) for creating an additional applied schema `NewtonKaminoPADMMSolverAPI` applied on the `NewtonKaminoSceneAPI` to facilitate potentially other solvers in the future.
